src/Mod/CAM/Path/Machine/models/machine.py

```diff
@@ -0,0 +1,1474 @@
+        return config
+
+    @classmethod
+    def create_3axis_config(cls):
+        """Create standard 3-axis XYZ configuration (no rotary axes)"""
+        config = cls("3-Axis XYZ Configuration")
+        config.add_linear_axis("X", FreeCAD.Vector(1, 0, 0))
+        config.add_linear_axis("Y", FreeCAD.Vector(0, 1, 0))
+        config.add_linear_axis("Z", FreeCAD.Vector(0, 0, 1))
+        config.description = "Standard 3-axis machine with no rotary axes"
+        # No rotary axes to add, no alignment axes to set
+        return config
+
+    def to_dict(self):
+        """Serialize configuration to dictionary for JSON persistence"""
+        # Build flattened axes structure
+        axes = {}
+
+        # Add linear axes from LinearAxis objects
+        for axis_name, axis_obj in self.linear_axes.items():
+            dir_vec = axis_obj.direction_vector
+            joint = [[dir_vec.x, dir_vec.y, dir_vec.z], [0, 0, 0]]
+
+            axes[axis_name] = {
+                "type": "linear",
+                "min": axis_obj.min_limit,
+                "max": axis_obj.max_limit,
+                "max_velocity": axis_obj.max_velocity,
+                "joint": joint,
+                "sequence": axis_obj.sequence,
+            }
+
+        # Add rotary axes
+        for axis_name, axis_obj in self.rotary_axes.items():
+            rot_vec = axis_obj.rotation_vector
+            joint = [[0, 0, 0], [rot_vec.x, rot_vec.y, rot_vec.z]]
+            axes[axis_name] = {
+                "type": "angular",
+                "min": axis_obj.min_limit,
+                "max": axis_obj.max_limit,
+                "max_velocity": axis_obj.max_velocity,
+                "joint": joint,
+                "sequence": axis_obj.sequence,
+                "prefer_positive": axis_obj.prefer_positive,
+            }
+
+        data = {
+            "freecad_version": self.freecad_version,
+            "machine": {
+                "name": self.name,
+                "manufacturer": self.manufacturer,
+                "description": self.description,
+                # "type": self.machine_type,
+                "units": self.configuration_units,
+                "axes": axes,
+                "spindles": [spindle.to_dict() for spindle in self.spindles],
+            },
+            "version": self.version,
+        }
+
+        # Add post-processor configuration
+        data["postprocessor"] = {
+            "file_name": self.postprocessor_file_name,
+            "args": self.postprocessor_args,
+            "motion_mode": self.motion_mode.value,
+        }
+
+        # Output options
+        data["output"] = {
+            "comments": self.output.comments,
+            "blank_lines": self.output.blank_lines,
+            "header": self.output.header,
+            "line_numbers": self.output.line_numbers,
+            "path_labels": self.output.path_labels,
+            "machine_name": self.output.machine_name,
+            "doubles": self.output.doubles,
```

Comment:
This 'doubles' property is supposed to be used in suppressing redundant gcode words.

IIRC, there were originally two controls that did this.  One suppressed redundant gcode commands (G0, G1, G81) and the other suppressed redundant words within those commands.  For example<

G1 X10 Y10 Z10
G1 X10 Y20 Z10

could be flattened to 
G1 X10 Y10 Z10
G1 Y10

Or it could be flattened all the way to 
G1 X10 Y10 Z10
Y10

Both are valid gcode but people have strong preferences about it.
I think we should have both
'output_double_commands' 
'output_double_parameters'

I'm open to better names for these things.